src/creatures/players/cast/cast_viewer.cpp

```diff
@@ -0,0 +1,1550 @@
+	}
+}
+
+void CastViewer::parseSendBosstiarySlots() {
+	if (m_owner) {
+		m_owner->parseSendBosstiarySlots();
+	}
+}
+
+void CastViewer::sendLootStats(std::shared_ptr<Item> item, uint8_t count) {
+	if (m_owner) {
+		m_owner->sendLootStats(item, count);
+	}
+}
+
+void CastViewer::sendUpdateSupplyTracker(std::shared_ptr<Item> item) {
+	if (m_owner) {
+		m_owner->sendUpdateSupplyTracker(item);
+	}
+}
+
+void CastViewer::sendUpdateImpactTracker(CombatType_t type, int32_t amount) {
+	if (m_owner) {
+		m_owner->sendUpdateImpactTracker(type, amount);
+	}
+}
+
+void CastViewer::openImbuementWindow(std::shared_ptr<Item> item) {
+	if (m_owner) {
+		m_owner->openImbuementWindow(item);
+	}
+}
+
+void CastViewer::sendMarketEnter(uint32_t depotId) {
+	if (m_owner) {
+		m_owner->sendMarketEnter(depotId);
+	}
+}
+
+void CastViewer::sendUnjustifiedPoints(const uint8_t &dayProgress, const uint8_t &dayLeft, const uint8_t &weekProgress, const uint8_t &weekLeft, const uint8_t &monthProgress, const uint8_t &monthLeft, const uint8_t &skullDuration) {
+	if (m_owner) {
+		m_owner->sendUnjustifiedPoints(dayProgress, dayLeft, weekProgress, weekLeft, monthProgress, monthLeft, skullDuration);
+
+		for (const auto &it : m_viewers) {
+			it.first->sendUnjustifiedPoints(dayProgress, dayLeft, weekProgress, weekLeft, monthProgress, monthLeft, skullDuration);
+		}
+	}
+}
+
+void CastViewer::sendModalWindow(const ModalWindow &modalWindow) {
+	if (m_owner) {
+		m_owner->sendModalWindow(modalWindow);
+	}
+}
+
+void CastViewer::sendResourceBalance(Resource_t resourceType, uint64_t value) {
+	if (m_owner) {
+		m_owner->sendResourceBalance(resourceType, value);
+	}
+}
+
+void CastViewer::sendOpenWheelWindow(uint32_t ownerId) {
+	if (m_owner) {
+		m_owner->sendOpenWheelWindow(ownerId);
+	}
+}
+
+void CastViewer::sendCreatureSay(std::shared_ptr<Creature> creature, SpeakClasses type, const std::string &text, const Position* pos) {
+	if (m_owner) {
+		m_owner->sendCreatureSay(creature, type, text, pos);
+
+		if (type == TALKTYPE_PRIVATE_FROM) {
+			return;
+		}
+
+		for (const auto &it : m_viewers) {
+			it.first->sendCreatureSay(creature, type, text, pos);
+		}
+	}
+}
+
+void CastViewer::disconnectClient(const std::string &message) const {
+	if (m_owner) {
+		m_owner->disconnectClient(message);
+	}
+}
+
+void CastViewer::addViewer(ProtocolGame_ptr client, bool spy) {
+	if (m_viewers.find(client) != m_viewers.end()) {
+		return;
+	}
+
+	std::stringstream message;
+	auto viewerId = m_viewers.size() + 1;
+	message << "Guest-" << viewerId;
+
+	m_viewers[client] = std::make_pair(message.str(), m_id);
+
+	if (!spy) {
+		sendChannelMessage("", message.str() + " has entered the cast.", TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+
+		if (m_viewers.size() > m_castLiveRecord) {
+			m_castLiveRecord = m_viewers.size();
+			sendChannelMessage("", "New record: " + std::to_string(m_castLiveRecord) + " people are watching your livestream now.", TALKTYPE_CHANNEL_O, CHANNEL_CAST);
+		}
+	}
+}
```

Comment:
What about this:
```suggestion
void CastViewer::addViewer(ProtocolGame_ptr client, bool spy) {
	if (m_viewers.find(client) != m_viewers.end()) {
		return;
	}

	auto viewerId = m_viewers.size() + 1;
	std::string guestString = fmt::format("Guest-{}", viewerId);

	m_viewers[client] = std::make_pair(guestString, m_id);

	if (!spy) {
		sendChannelMessage("", fmt::format("{} has entered the cast.", guestString), TALKTYPE_CHANNEL_O, CHANNEL_CAST);

		if (m_viewers.size() > m_castLiveRecord) {
			m_castLiveRecord = m_viewers.size();
			sendChannelMessage("", fmt::format("New record: {} people are watching your livestream now.", std::to_string(m_castLiveRecord), TALKTYPE_CHANNEL_O, CHANNEL_CAST);
		}
	}
}
```
?